xmodule/modulestore/__init__.py

```diff
@@ -313,13 +316,52 @@ def send_pre_publish_signal(self, bulk_ops_record, course_id):
         if signal_handler and bulk_ops_record.has_publish_item:
             signal_handler.send("pre_publish", course_key=course_id)
 
+    def create_catalog_data_for_signal(self, course_id: CourseKey) -> CourseCatalogData:
+        """
+        Creates a CourseCatalogData object from a course id by fetching from the module store
+
+        Arguments:
+            course_id: id of the course to process
+        """
+        # Most of this is copied from CourseOverview. We cannot import it
+        # because it would lead to circular imports
+        course_from_store = self.get_course(course_id)
+        number = number_for_course_location(course_from_store.location)
+        catalog_visibility = course_from_store.catalog_visibility
+
+        # see CourseDetail.fetch_about_attribute
+        effort_usage_key = course_id.make_usage_key('about', 'effort')
+        try:
+            effort = self.get_item(effort_usage_key).data
+        except ItemNotFoundError:
+            effort = None
+        return CourseCatalogData(
+            course_key=course_id.for_branch(None),
+            name=course_from_store.display_name,
+            org=course_from_store.location.org,
```

Comment:
Raised a question about whether we want these fields: https://github.com/openedx/openedx-events/issues/72#issuecomment-1212177933